### PR TITLE
APIDiff.swift: Address warning  by using task group instead of semaphore

### PR DESCRIPTION
**Pull request by valeriyvan**

### CODE REVIEW

File: Sources/Commands/PackageCommands/APIDiff.swift [hunk 1]
```diff
@@ -119,38 +119,41 @@ struct APIDiff: AsyncSwiftCommand {
...
+                    } catch {
+                        swiftCommandState.observabilityScope.emit(error: "failed to compare API to baseline", underlyingError: error)
                     }
-                } catch {
-                    swiftCommandState.observabilityScope.emit(error: "failed to compare API to baseline", underlyingError: error)
+                    return nil
                 }
-                semaphore.signal()
             }
+            let results = ThreadSafeArrayStore<SwiftAPIDigester.ComparisonResult>()
```
[the remark this answers is not in the document]
Comment (valeriyvan) [reply Sources/Commands/PackageCommands/APIDiff.swift]: done

**Comment by rauhul** [comment]
Could we factor this into a more reusable component, like this?

[Sequence+ParallelMap.swift.txt](https://github.com/user-attachments/files/16971122/Sequence%2BParallelMap.swift.txt)

**Comment by valeriyvan** [comment]
> Could we factor this into a more reusable component, like this?

Not in this PR. Open an issue if you think it does worth it.

**Comment by MaxDesiatov** [comment]
You'll have to add `import _Concurrency` to the file you've changed to fix CMake builds and unblock CI.

**Review by MaxDesiatov — CHANGES_REQUESTED** [review]
You'll have to add import _Concurrency to the file you've changed to fix CMake builds and unblock CI.

**Comment by valeriyvan** [comment]
> You'll have to add import _Concurrency to the file you've changed to fix CMake builds and unblock CI.

done

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by MaxDesiatov — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**